### PR TITLE
DROID-3185 File as Object | New layout ui, part 1

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt [hunk 1]
```diff
@@ -530,6 +542,12 @@ class BlockAdapter(
                     ItemBlockMediaErrorBinding.inflate(inflater, parent, false)
                 )
             }
+            HOLDER_OPEN_FILE -> {
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt:545]: HOLDER_BUTTON_OPEN_FILE

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt [hunk 2]
```diff
@@ -530,6 +542,12 @@ class BlockAdapter(
...
+                OpenFile(ItemBlockOpenFileBinding.inflate(inflater, parent, false))
+            }
+            HOLDER_OPEN_IMAGE -> {
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt:548]: idem

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/holders/other/Title.kt [hunk 3]
```diff
@@ -555,4 +555,33 @@ sealed class Title(view: View) : BlockViewHolder(view), TextHolder {
...
+            item: BlockView.Title.File,
+        ) {
+            super.bind(
+                item = item,
+                onCoverClicked = {}
+            )
+            icon.setIcon(item.icon)
+        }
+
+        override fun applyTextColor(item: BlockView.Title) {}
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/holders/other/Title.kt:584]: let's add comment: "// Do nothing"

File: presentation/src/main/java/com/anytypeio/anytype/presentation/editor/editor/model/BlockView.kt [hunk 4]
```diff
@@ -1268,6 +1296,28 @@ sealed class BlockView : ViewType {
         override fun getViewType(): Int = HOLDER_FEATURED_RELATION
     }
 
+    sealed class OpenFile(
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/editor/editor/model/BlockView.kt:1299]: `ButtonOpenFile`

**Review by uburoiubu — APPROVED** [review]

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt (hunk 1 above)
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt:545]: done

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt (hunk 2 above)
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/BlockAdapter.kt:548]: done

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/holders/other/Title.kt (hunk 3 above)
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/features/editor/holders/other/Title.kt:584]: done

File: presentation/src/main/java/com/anytypeio/anytype/presentation/editor/editor/model/BlockView.kt (hunk 4 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/editor/editor/model/BlockView.kt:1299]: done

**Pull request merged**